### ISSUE TITLE
Crash after few minutes

### ISSUE THREAD

**Issue by HellDunkel** [ziicreater/socketcluster-client-ue4#3]
I set up Socketcluster in UE 4.23 as below and publish the speed of a moving object at every tick. After 3 min (more or less) i get a crash in "_handleEventAckTimeout".
To be honest- i have little idea of what i am doing here. The settings don't mean much to me as i am new to SocketCluster.

![socketcluster_Connect](https://user-images.githubusercontent.com/20302808/68430775-fe449200-01b0-11ea-8efd-b26cb770e174.JPG)
![socketcluster_Publish](https://user-images.githubusercontent.com/20302808/68430793-0997bd80-01b1-11ea-9ee4-996297f36b40.JPG)

**Comment by SloppyShovel** [comment]
Dear @HellDunkel 

I was able to reproduce the issue, we are looking into it

I will keep you updated.

**Comment by HellDunkel** [comment]
Hello @inQonsole,  Any idea on  why this i happening? Thanks for looking at it!

**Comment by SloppyShovel** [comment]
Yes @HellDunkel , the issue is every call event has a timer linked to them so that when the server does not respond it times out, the problem now is, the timers and callbacks are added to a list in no specific order, so with Event Tick, you are adding so many to the list as such a fast rate, the event object gets deleted before the timer object is cleared, so we are planning on replacing the list with a key-value list where the key is the call id so it doesn't run out of sync,

**Comment by HellDunkel** [comment]
@inQonsole Interesting. I was suspecting that sending at tick rate was sort of a misuse but it seemed to do the job well and quick enough. I just need this for a prototype setup- nothing that would ship. The plan was going to send a few more signals at tick rate. Would you recommend a different approach here or do you see SocketCluster suitable for this at all?

**Comment by SloppyShovel** [comment]
@HellDunkel It is a misuse in terms of you don't wanna do anything like that on tick rate for sure, but still something that needs to be fixed, so in that sense, it's very good you tested and brought this up, there should be no limit in how fast you can send data to socketcluster, my advice would send the data on the object movement with a delay, so only when the object moves you should update for example, but this issue is in our issue tracker.

**Comment by HellDunkel** [comment]
@inQonsole  I actually need those signals at a very high frequency (every frame). Let me rephrase the question: Would you recommend a C++ implementation i.e. using UE4s FSocket instead of using a webserver? The latter has some nice benefits to my usecase but i am skeptical it can keep up with performance. Imagine sending ~100 float values at every tick.
Thank you- i really appreciate your help.

**Comment by SloppyShovel** [comment]
@HellDunkel I don't think it's realistic for an every frame update but oke if it's needed, yes I would implant it in C++ it will give some performance benefits if using the FSocket is going to give better performance, I do not know, the socket cluster client is directly written on top of libwebsockets same as UE4 uses, so i don't think you going get any performance benefits out of that but you could try.

When the timer function has been fixed there wouldn't be an issue sending any about of data at tick rate for socketcluster.

**Comment by HellDunkel** [comment]
Thank you so much- looking forward to the fix!

**Comment by HellDunkel** [comment]
@inQonsole Any update here? :)

**Comment by SloppyShovel** [comment]
@HellDunkel I just returned from a business trip, i will check with the team if anybody had time to work on this, i let you know

**Comment by HellDunkel** [comment]
@inQonsole  still looking forward to this! ;)

**Comment by SloppyShovel** [comment]
@HellDunkel Sorry for the delay! I got really busy, the issue has been found and fixed, v0.5 has the issue fixed, it will take a couple of days before the fix will be available on the marketplace.

**Comment by HellDunkel** [comment]
@inQonsole That is great news- thank you so much! :)
